hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -1383,6 +1404,22 @@ int QuectelNcpClient::modemPowerOff() const {
     return SYSTEM_ERROR_NONE;
 }
 
+int QuectelNcpClient::modemSoftPowerOff() {
+    if (modemPowerState()) {
+        LOG(TRACE, "Powering modem off using AT command");
```

Comment:
We should probably check for `ready_` here before attempting to execute AT commands. `parser_.execCommand()` will fail anyway, but it might be cleaner to even not attempt to do that.

Reply:
Same for SaraNcpClient.